src/github.com/getlantern/lantern-android/client/client.go

```diff
@@ -124,141 +120,17 @@ func (client *Client) pollConfiguration() {
-	}
-
-	log.Printf("Starting proxy server at %s...", addr)
-
-	go client.pollConfiguration()
-
-	return httpServer.Serve(client.ln)
-}
-
-func targetQOS(req *http.Request) int {
-	requestedQOS := req.Header.Get(httpXFlashlightQOS)
-	if requestedQOS != "" {
-		rqos, err := strconv.Atoi(requestedQOS)
-		if err == nil {
-			return rqos
-		}
-	}
-	return 0
-}
-
-// intercept intercepts an HTTP CONNECT request, hijacks the underlying client
-// connetion and starts piping the data over a new net.Conn obtained from the
-// given dial function.
-func (client *Client) intercept(resp http.ResponseWriter, req *http.Request) {
-	if req.Method != httpConnectMethod {
-		panic("Intercept used for non-CONNECT request!")
-	}
-
-	// Hijack underlying connection
-	clientConn, _, err := resp.(http.Hijacker).Hijack()
-	if err != nil {
-		respondBadGateway(resp, fmt.Sprintf("Unable to hijack connection: %s", err))
-		return
-	}
-	defer clientConn.Close()
-
-	addr := hostIncludingPort(req, 443)
-
-	// Establish outbound connection
-	connOut, err := client.getBalancer().DialQOS("tcp", addr, targetQOS(req))
-	if err != nil {
-		respondBadGateway(clientConn, fmt.Sprintf("Unable to handle CONNECT request: %s", err))
-		return
-	}
-	defer connOut.Close()
-
-	// Pipe data
-	pipeData(clientConn, connOut, req)
 }
 
 // Stop is currently not implemented but should make the listener stop
 // accepting new connections and then kill all active connections.
-func (client *Client) Stop() error {
+func (client *MClient) Stop() error {
 	log.Printf("Stopping proxy server...")
-	return client.ln.Close()
-}
-
-func respondBadGateway(w io.Writer, msg string) error {
-	log.Printf("Responding BadGateway: %v", msg)
-	resp := &http.Response{
-		StatusCode: http.StatusBadGateway,
-		ProtoMajor: 1,
-		ProtoMinor: 1,
-	}
-	err := resp.Write(w)
-	if err == nil {
-		_, err = w.Write([]byte(msg))
-	}
-	return err
-}
-
-// hostIncludingPort extracts the host:port from a request.  It fills in a
-// a default port if none was found in the request.
-func hostIncludingPort(req *http.Request, defaultPort int) string {
-	_, port, err := net.SplitHostPort(req.Host)
-	if port == "" || err != nil {
-		return req.Host + ":" + strconv.Itoa(defaultPort)
-	}
-	return req.Host
-}
-
-// pipeData pipes data between the client and proxy connections.  It's also
-// responsible for responding to the initial CONNECT request with a 200 OK.
-func pipeData(clientConn net.Conn, connOut net.Conn, req *http.Request) {
-	// Start piping to proxy
-	go io.Copy(connOut, clientConn)
-
-	// Respond OK
-	err := respondOK(clientConn, req)
-	if err != nil {
-		log.Printf("Unable to respond OK: %s", err)
-		return
-	}
-
-	// Then start coyping from out to client
-	io.Copy(clientConn, connOut)
-}
-
-func respondOK(writer io.Writer, req *http.Request) error {
-	defer req.Body.Close()
-	resp := &http.Response{
-		StatusCode: http.StatusOK,
-		ProtoMajor: 1,
-		ProtoMinor: 1,
-	}
-	return resp.Write(writer)
+	return client.hqfd.Close()
```

Comment:
Yes, it's intended to. I'm basically just supporting the Stop functionality you provided previously.